nucypher/blockchain/economics.py

```diff
@@ -167,6 +167,31 @@ def staking_deployment_parameters(self) -> Tuple[int, ...]:
         )
         return tuple(map(int, deploy_parameters))
 
+    def token_supply_function(self, at_period: int) -> int:
+        if at_period < 0:
+            raise ValueError("Period must be a positive integer")
+
+        with localcontext() as ctx:
+            ctx.prec = self._precision
+
+            # Eq. 3 of the mining paper
+            t = Decimal(at_period)
+            S_0 = self.erc20_initial_supply
+            i_0 = 1
+            I_0 = i_0 * S_0  # in 1/years
+            T_half = self.token_halving  # in years
+            T_half_in_days = T_half * 365
+
+            S_t = S_0 + I_0 * T_half * (1 - 2**(-t / T_half_in_days)) / LOG2
+            return int(S_t)
+
+    def cumulative_rewards_at_period(self, at_period: int) -> int:
+        return self.token_supply_function(at_period) - self.erc20_initial_supply
+
+    def rewards_during_period(self, period: int) -> int:
```

Comment:
``` suggestion
    def rewards_at_period(self, period: int) -> int:
```

Reply:
This reward is not exactly "at" period, but during period - difference between current and previous reward